frontend/src/components/test-history.js

```diff
@@ -33,334 +31,317 @@ import {
+  const [isEmpty, setIsEmpty] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [onlyFailures, setOnlyFailures] = useState(false);
+  const [historySummary, setHistorySummary] = useState();
+  const [dropdownSelection, setDropdownSelection] = useState('1 Week');
+  const [filtersState, setFiltersState] = useState({});
 
-    // lazy-load the result view so we don't have to make a bunch of artifact requests
-    if (isOpen) {
-      rows[rowIndex + 1].cells = [{
-        title: <ResultView hideTestHistory={true} hideSummary={false} hideTestObject={false} testResult={rows[rowIndex].result}/>
-      }];
+  useEffect(() => {
+    let env_filter = {};
+    if (testResult?.env) {
+      env_filter['env'] = {
+        op: 'eq',
+        val: testResult.env
+      };
     }
-    rows[rowIndex].isOpen = isOpen;
-    this.setState({rows});
-  }
+    let time_filter = {};
+    if (testResult?.start_time) {
+      // default to filter only from 1 weeks ago to the most test's start_time.
+      time_filter['start_time'] = {
+        op: 'gt',
+        val: new Date(new Date(testResult?.start_time).getTime() - (0.25 * 30 * 86400 * 1000)).toISOString()
+      };
+    }
+    setFiltersState(
+      {
+        ...filters,
+        'result': {
+          op: 'in',
+          val: 'passed;skipped;failed;error;xpassed;xfailed'
+        },
+        'test_id': {
+          op: 'eq',
+          val: testResult?.test_id
+        },
+        'component': {
+          op: 'eq',
+          val: testResult?.component
+        },
+        ...time_filter,
+        ...env_filter
+      }
+    );
 
-  setPage = (_event, pageNumber) => {
-    this.setState({page: pageNumber}, () => {
-      this.getResultsForTable();
-    });
-  };
+  }, [testResult, filters]);
 
-  pageSizeSelect = (_event, perPage) => {
-    this.setState({pageSize: perPage}, () => {
-      this.getResultsForTable();
-    });
+  const onCollapse = (event, rowIndex, isOpen) => {
+    // lazy-load the result view so we don't have to make a bunch of artifact requests
+    // TODO with ResultView moving tab rendering and artifact fetching into ArtifactTab, this may not be necessary anymore
+    setRows(rows.map((row, index) => {
+      if (index === (rowIndex + 1)){
+        return ({
+          ...row,
+          'cells': [{
+            title: <ResultView defaultTab='summary' hideTestHistory={true} hideSummary={false} hideTestObject={false} testResult={rows[rowIndex].result}/>
+          }]});
+      } else if (index === rowIndex) {
+        return({
+          ...row,
+          'isOpen': isOpen
+        });
+      } else {
+        return(row);
+      }
+    }));
   };
 
-  updateFilters (name, operator, value, callback) {
-    let filters = this.state.filters;
+  const updateFilters = useCallback((name, operator, value) => {
+    let updatedfilters = {...filtersState};
```

Comment:
Nitpick: Could be const since we are not re-assigning object but changing it's props

Reply:
thanks for these! good to be consistent, and you're right only the props are changing but the type doesn't.